components/icu/examples/tui.rs

```diff
@@ -37,18 +40,18 @@ fn main() {
     println!("User: {user_name}");
 
     {
-        let dtf = TypedZonedDateTimeFormatter::<Gregorian>::try_new(
-            &locale,
-            DateTimeFormatterOptions::default(),
-            TimeZoneFormatterOptions::default(),
-        )
-        .expect("Failed to create TypedDateTimeFormatter.");
-        let today_date = DateTime::try_new_gregorian_datetime(2020, 10, 10, 18, 56, 0).unwrap();
-        let today_tz = CustomTimeZone::try_from_str("Z").unwrap(); // Z refers to the utc timezone
+        let dtf = TypedNeoFormatter::<
+            Gregorian,
+            NeoYearMonthDayHourMinuteSecondTimeZoneGenericShortMarker,
+        >::try_new(&locale, NeoSkeletonLength::Medium.into())
+        .expect("Failed to create zoned datetime formatter.");
+        let date = Date::try_new_gregorian_date(2020, 10, 10).unwrap();
+        let time = Time::try_new(18, 56, 0, 0).unwrap();
+        let zone = CustomTimeZone::utc();
 
-        let formatted_dt = dtf.format(&today_date, &today_tz);
+        let formatted_dt = dtf.format(&CustomZonedDateTime { date, time, zone });
 
-        println!("Today is: {formatted_dt}");
+        println!("Today is: {}", LossyWrap(formatted_dt));
```

Comment:
I think this is a bit quirky for Rust users. I'm wondering if we can impl Writeable for formatted_dt by default to avoid having to ask for this wrapper to use println.

Reply:
I'm open to that, implementing both Writeable and TryWriteable, although we previously decided that we wouldn't generally do that and so it was okay to have name clashes on the functions such as writeable_length_hint

Reply:
Or maybe we could impl TryWriteable and Display without Writeable?

Reply:
Oh, I kinda like that!

Reply:
remind me what the error path is?

Reply:
If the input fields don't match the output fields. There exist ways to abuse the API and get that behavior, but it shouldn't happen with how we write the docs test code. You wrote much of the error handling fallback code.